feat: Remove PlaceholderRelationSerializer and add Placeholder content instead

fsbraun opened the pull request:
This PR adds the Placeholder content directly to placeholder relation fields, e.g., for the page detail endpoint:

```
GET /api/en/pages/?html=true

HTTP 200 OK
Allow: GET, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "title": "test page",
    ...,
    "placeholders": [
        {
            "slot": "Page Content",
            "label": "Page Content",
            "language": "en",
            "content": [
                {
                    "plugin_type": "TextPlugin",
                    "body": "<p>No need to fetch placeholder content in another request response cycle</p>",
                    "json": {
                        "type": "doc",
                        "content": [
                            {
                                "type": "paragraph",
                                "attrs": {
                                    "textAlign": null
                                },
                                "content": [
                                    {
                                        "type": "text",
                                        "text": "No need to fetch placeholder content in another request response cycle"
                                    }
                                ]
                            }
                        ]
                    },
                    "rte": "tiptap"
                }
            ],
            "details": "http://127.0.0.1:8000/api/en/placeholders/10/11/Page%20Content/?html=true",
            "html": "<p>No need to fetch placeholder content in another request response cycle</p>"
        }
    ]
}
```


Tests need to be adjusted.

## Summary by Sourcery

Provide placeholder content, HTML, and details URL directly in page responses by consolidating placeholder serialization into a single serializer

New Features:
- Include placeholder content and HTML directly in the page detail endpoint response

Enhancements:
- Remove PlaceholderRelationSerializer and use PlaceholderSerializer for placeholders in pages
- Move plugin rendering and HTML fetching into PlaceholderSerializer.to_representation
- Add "details" URL field to PlaceholderSerializer and preserve original query parameters in its URL generation

Tests:
- Update tests to expect direct placeholder content, HTML, and details fields

fsbraun commented on djangocms_rest/serializers/placeholders.py:
```diff
@@ -12,71 +12,53 @@ class PlaceholderSerializer(serializers.Serializer):
+    def to_representation(self, instance):
+        instance.label = instance.get_label()
+        instance.language = self.language
+        instance.details = self.get_details(instance)
```
@metaforx What do you think about this? Maybe we change the pattern.

metaforx commented on tests/types.py:
```diff
@@ -29,15 +29,18 @@
-    "content_type_id": int,
-    "object_id": int,
+
+PLACEHOLDER_FIELD_TYPES = {
```
relates to missing `content`

fsbraun commented:
@metaforx I fixed the schema issue and added test for the schema.

I do not think that generating the content is too traffic heavy: In the end it's a page worth of content. 

Once we have added the path feature to the menus, would the menus provide enough meta info on the site structure and also on the pages? We also have `pages-list` and  `pages-tree` for page-only meta info.

I'll leave it for you to decide if we add a set of `page-meta` endpoints wich would not contain the content. Let me know.
